repository.go

```diff
@@ -133,14 +162,21 @@ func (r *repository) fetch() error {
 	r.etag = resp.Header.Get("Etag")
 	r.segments = featureResp.SegmentsMap()
 	r.options.storage.Reset(featureResp.FeatureMap(), true)
+	r.successfulFetch()
 	r.Unlock()
 	return nil
 }
 
-func statusIsOK(resp *http.Response) error {
+func (r *repository) statusIsOK(resp *http.Response) error {
 	s := resp.StatusCode
 	if 200 <= s && s < 300 {
 		return nil
+	} else if s == 401 || s == 403 || s == 404 {
+		r.configurationError()
+		return fmt.Errorf("%s %s returned status code %d your SDK is most likely misconfigured, backing off to maximum (%f times our interval)", resp.Request.Method, resp.Request.URL, s, r.maxSkips)
+	} else if s == 429 || s >= 500 {
+		r.backoff()
+		return fmt.Errorf("%s %s returned status code %d, backing off (%f times our interval)", resp.Request.Method, resp.Request.URL, s, r.errors)
```

Comment:
Here we use the default HTTP codes, whereas in metrics we use the HTTP helpers. We should be consistent.

Reply:
Agreed, I'll update to using helpers here as well

Reply:
Fixed in #7c8994d